src/views/studio/CreateEditChannelView/CreateEditChannelView.tsx

```diff
@@ -0,0 +1,271 @@
+import { ChannelCover, FormField, Select, SelectedItem, HeaderTextField, Tooltip } from '@/shared/components'
+
+import { transitions } from '@/shared/theme'
+import {
+  StyledTitleSection,
+  TitleContainer,
+  StyledFormField,
+  StyledTextarea,
+  StyledActionBarTransaction,
+  StyledAvatar,
+  InnerFormContainer,
+} from './CreateEditChannelView.style'
+import { Header, SubTitle, SubTitlePlaceholder, TitlePlaceholder } from '@/views/consumer/ChannelView/ChannelView.style'
+
+import { requiredValidation, textFieldValidation } from '@/utils/formValidationOptions'
+import { formatNumberShort } from '@/utils/number'
+
+const isPublicSelect: SelectedItem[] = [
+  { name: 'Public (Channel will appear in feeds)', value: true },
+  { name: 'Unlisted', value: false },
+]
+
+const FEE = 0
+
+type ImageAsset = {
+  url: string | null
+  blob: Blob | null
+}
+type Inputs = {
+  channelName?: string
+  description?: string
+  isPublic: SelectedItem
+  selectedLanguage: SelectedItem
+  avatar: ImageAsset
+  cover: ImageAsset
+}
+
+type CreateEditChannelViewProps = {
+  newChannel?: boolean
+}
+
+const CreateEditChannelView: React.FC<CreateEditChannelViewProps> = ({ newChannel }) => {
+  // TODO Add hook for fetching currently active channel
+  const id = '1494'
+  const { channel, loading, error } = useChannel(id, { skip: newChannel })
+
+  const {
+    register,
+    handleSubmit: useHandleSubmit,
+    control,
+    setValue,
+    getValues,
+    formState: { isDirty },
+    reset,
+    watch,
+    errors,
+  } = useForm<Inputs>({
+    defaultValues: {
+      avatar: { url: null, blob: null },
+      cover: { url: null, blob: null },
+      channelName: '',
+      description: '',
+      selectedLanguage: languages[0],
+      isPublic: isPublicSelect[0],
+    },
+  })
+
+  const avatarDialogRef = useRef<ImageCropDialogImperativeHandle>(null)
+  const coverDialogRef = useRef<ImageCropDialogImperativeHandle>(null)
+
+  const [avatarImageUrl, setAvatarImageUrl] = useState<string | null>(null)
+  const [coverImageUrl, setCoverImageUrl] = useState<string | null>(null)
+  const [isActionBarActive, setActionBarActive] = useState(false)
+
+  useEffect(() => {
+    if (loading || newChannel) {
+      return
+    }
+    if (!channel) {
+      return
+    }
+    const { avatarPhotoUrl, coverPhotoUrl, handle, description, isPublic, language } = channel
+    const foundLanguage = languages.find(({ name }) => name === language?.name)
+    reset({
+      ...getValues(),
+      avatar: { blob: null, url: avatarPhotoUrl },
+      cover: { blob: null, url: coverPhotoUrl },
+      channelName: handle,
+      description: description || '',
+      isPublic: isPublic ? isPublicSelect[0] : isPublicSelect[1],
+      selectedLanguage: foundLanguage || languages[0],
+    })
+  }, [channel, getValues, loading, newChannel, reset])
+
+  useEffect(() => {
+    setAvatarImageUrl(watch().avatar.url)
```

Comment:
Why do we need those? We already call those setters in `handleAvatarConfirm` and `handleCoverConfirm`

Reply:
This hook will be called a lot of times